README.md

```diff
@@ -145,7 +161,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+*See attatched program
```

Comment:
Thanks for the tests. They aren't really proper automated tests, though, because they rely on a human to read each line. An automated test should know what the expected answer is, and report when the result differs.

# -2

Your program implements the state machine entirely in code: I was looking for a table based approach, something like this:

~~~ cpp
state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~~

The program also doesn't parse my grammar: if I feed it my sentences with smelly dogs and lazy cats, it doesn't recognize them.

# -7